modules/bitgo/src/v2/coins/sol.ts

```diff
@@ -161,10 +162,23 @@ export class Sol extends BaseCoin {
     }
     if (txParams.recipients) {
       for (const recipients of txParams.recipients) {
-        totalAmount = totalAmount.plus(recipients.amount);
+        // totalAmount based on each token
+        const assetName = recipients.tokenName || this.getChain();
+        const amount = totalAmount[assetName] || new BigNumber(0);
+        totalAmount[assetName] = amount.plus(recipients.amount);
       }
 
-      if (!totalAmount.isEqualTo(explainedTx.outputAmount)) {
+      // total output amount from explainedTx
+      const explainedTxTotal: Record<string, BigNumber> = {};
+
+      for (const output of explainedTx.outputs) {
+        // total output amount based on each token
+        const assetName = output.tokenName || this.getChain();
+        const amount = explainedTxTotal[assetName] || new BigNumber(0);
+        explainedTxTotal[assetName] = amount.plus(output.amount);
+      }
+
+      if (JSON.stringify(explainedTxTotal) !== JSON.stringify(totalAmount)) {
```

Comment:
same here

Reply:
Same issue with above, because of `explainTransaction`, the output can have `tokenName` field undefined where the cause is from bitgo-microservice somewhere during prebuildTransaction

Reply:
is this still the case?

Reply:
yes, since below is updating `parseTransaction` while here is using `explainTransaction`
can't use `parseTransaction` since it only return inputs and outputs value while `verifyTransaction` still need to check Nonce etc

Reply:
I'm not seeing the tests fail here on this one. Seems to pass when I use 
```
if (!_.isEqual(explainedTxTotal, totalAmount))
```

We should avoid unnecessarily serializing objects just to check for equality, there's usually a better way to accomplish that.

Reply:
Ohh got it, my bad